run.go

```diff
@@ -326,38 +329,96 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := []string{}
+	seen := map[string]string{}
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			result = append(result, val)
+			seen[val] = val
+		}
+	}
+	return result
+}
+
+// Checks if outputVariables are defined in terms of other outputVariables.
```

Comment:
The convention is to start function comments with the name of the function. E.g.: 

checkForDerivitives creates a slice of unique ...